### DISCUSSION TITLE
Make autofocus optional for Flux modals

### DISCUSSION THREAD

**Discussion by thijskuilman** [livewire/flux#796] [Feature Requests]
I notice that Flux modals tries to autofocus on the first input (?), which in most cases is a nice feature. However, sometimes I would like to disable this autofocus behaviour and I do not see an option for that.

I would like to see an optional autofocus property for Flux modals. I think Filament is a nice example for this:

<img width="635" alt="image" src="https://github.com/user-attachments/assets/2eda2864-0ae8-4672-b287-0fabd9efde8f">

**Comment by ju5t** [comment]
I'm almost inclined to call this an issue. Because when there's pagination inside a modal it focusses on that:

![CleanShot 2024-12-05 at 10 22 06](https://github.com/user-attachments/assets/aef77069-b678-49ff-ba2c-7e2e97dc354a)

A user wouldn't expect that to be focussed, so it kind of looks like that's the active page now.

**Reply by edwardkarlsson** [reply]
Agreed, even if there is nothing focusable within the modal, it focuses on the close button... which for me, and maybe quite a few others, isn't what I want.

**Reply by joshhanley** [reply]
Please see details in issue #854

**Comment by s-tinga** [comment]
^^ I'd like to highlight this issue too! I have modals with a heading, text, then an action button and because of this autofocus functionality the user is automatically focused on the button the end of the modal. This is great for your average site user, but it means screenreaders get jumped to the end of the content and then have to move backwards in order to read everything, then move forward again to get to the action button. It'd be much nicer for screenreaders if the autofocus either could be entirely turned off or set to the first element in a modal instead.

**Reply by joshhanley** [reply]
@stuininga1 thanks for all your comments so far about accessibility! Please keep them coming. Can I ask (again) that you open this as an issue and explain there the impact of this on screenreaders? That way it will get it in front of Caleb so he can have a look 🙂

**Comment by binaryfire** [comment]
Agreed, there needs to be a way of disabling this. Most of the time it's fine but there are definitely times where it isn't desirable.

**Reply by jeffchown** [reply]
@binaryfire This issue is now here: https://github.com/livewire/flux/issues/854

**Comment by joshhanley** [comment]
Going to close this discussion as there is an open issue #854 for this.
